pkg/controller/admission/merge_aerospikeconfig.go

```diff
@@ -0,0 +1,158 @@
+package admission
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func merge(base, patch map[string]interface{}) (map[string]interface{}, error) {
+	if len(patch) == 0 {
+		return base, nil
+	}
+
+	res := map[string]interface{}{}
+
+	for key, patchValue := range patch {
+		baseValue, ok := base[key]
+		// value was added
+		if !ok {
+			res[key] = patchValue
+			continue
+		}
+		// If types have changed, replace completely
+		if reflect.TypeOf(baseValue) != reflect.TypeOf(patchValue) {
+			res[key] = patchValue
+			continue
+		}
+		// Special check for key "storage-engine"
+		// Check value type and replace if it's type has changed
+		if key == "storage-engine" && isStorageEngineTypeChanged(baseValue, patchValue) {
+			res[key] = patchValue
+			continue
+		}
+		// Types are the same, compare values
+		val, err := handleValues(baseValue, patchValue)
+		if err != nil {
+			return nil, err
+		}
+		res[key] = val
+	}
+	// Now add other values
+	for k, v := range base {
+		if _, found := patch[k]; !found {
+			res[k] = v
+		}
+	}
+	return res, nil
+
+}
+
+func isStorageEngineTypeChanged(base, patch interface{}) bool {
+	// storage-engine = "memory"
+	_, ok1 := base.(string)
+	_, ok2 := patch.(string)
+
+	if ok1 && ok2 {
+		return false
+	}
+
+	baseMap, ok1 := base.(map[string]interface{})
```

Comment:
What about the change from memory to file or device?

Reply:
That is type change. Below condition should cover for that

		// If types have changed, replace completely
		if reflect.TypeOf(baseValue) != reflect.TypeOf(patchValue) {
			res[key] = patchValue
			continue
		}